### PR TITLE
[Custom Fields] Minor refactoring to simplify models definitions

### PR DESCRIPTION
**Pull request by hichamboushaba**
## Description
This PR has two main changes:
1. It attempts to simplify the model definitions for custom fields screens, instead of using two models that serve the same purpose on the same layer `CustomFieldViewModel` and `CustomFieldUI`, this PR removes the second one and consolidates the logic on the first one.
2. It marks the not-used functions as private and updates the tests accordingly; this is mainly for the functions `addField` and `editField`, which were referenced only privately from `saveField` and not directly from the UI. This change will ensure that our tests match the actual behavior of the UI, and also allows for an easier refactoring in the future, as the tests will be more rigid.

@hafizrahman this is not urgent given that it doesn't impact the functionality, please review it when you have some free time.

## Steps to reproduce
Hopefully unit tests should catch any potential issues, but please smoke test the custom fields feature.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldEditorViewModel.swift [hunk 1]
```diff
@@ -42,17 +42,20 @@ final class CustomFieldEditorViewModel: ObservableObject {
...
          onSave: @escaping (String, String) -> Void,
          onDelete: (() -> Void)? = nil) {
-        self.key = key
-        self.value = isJsonField ? value.prettyPrint() : value
-        self.initialKey = key
-        self.initialValue = isJsonField ? value.prettyPrint() : value
-        self.isReadOnlyValue = isJsonField
+        self.key = customField?.key ?? ""
+        self.initialKey = customField?.key ?? ""
+        let value = if customField?.isJson ?? false, let value = customField?.value {
```
Comment (hafizrahman) [inline WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldEditorViewModel.swift]: Maybe this shows the intent better? 

```suggestion
        let value = if customField?.isJson == true, let value = customField?.value {
```

**Review by hafizrahman — APPROVED** [review]
Thanks, these are great improvements to the code. I smoke tested the feature in Order and Products and they're still working fine.

Left a non-blocking code but pre-approving.

File: WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldEditorViewModel.swift (hunk 1 above)
Comment (hichamboushaba) [reply WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldEditorViewModel.swift]: Thanks, applied.
For context, I looked online, and there doesn't seem to be a consensus on the best syntax here unlike Kotlin, so I'm applying your suggestion if you think it's easier to read.

**Pull request merged**